examples/crowdfunding/test/crowdfundingTest.js

```diff
@@ -0,0 +1,252 @@
+        sign: SignType.LogicSignature,
+        fromAccount: { addr: escrow.address },
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        lsig: escrow,
+        payFlags: { closeRemainderTo: creator.address }
+      }
+    ];
+    // execute transaction: Expected to be rejected by logic because goal is not reached
+    try {
+      await runtime.executeTx(txGroup, program, []);
+    } catch (e) {
+      console.warn(e);
+    }
+  });
+
+  it('donor should be able to reclaim if goal is not met', async () => {
+    const appArgs = [stringToBytes('reclaim')];
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    const txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs,
+        accounts: [escrow.address] //  AppAccounts
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: donor.address,
+        amountMicroAlgos: 300000,
+        lsig: escrow,
+        payFlags: {}
+      }
+    ];
+
+    syncAccounts();
+    const donorBalance = donor.balance();
+    await runtime.executeTx(txGroup, program, []);
+
+    syncAccounts();
+    assert.equal(escrow.balance(), 300000);
+    assert.equal(donor.balance(), donorBalance + 300000);
+  });
+
+  it('should claim if goal is reached', async () => {
+    let appArgs = [stringToBytes('donate')];
+    const donationAmount = 7000000;
+
+    // Atomic Transaction (Stateful Smart Contract call + Payment Transaction)
+    let txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.SecretKey,
+        fromAccount: donor.account,
+        toAccountAddr: escrow.address,
+        amountMicroAlgos: donationAmount,
+        payFlags: {}
+      }
+    ];
+    // execute transaction
+    await runtime.executeTx(txGroup, program, []);
+
+    appArgs = [stringToBytes('claim')];
+    txGroup = [
+      {
+        type: TransactionType.CallNoOpSSC,
+        sign: SignType.SecretKey,
+        fromAccount: creator.account,
+        appId: applicationId,
+        payFlags: {},
+        appArgs: appArgs
+      },
+      {
+        type: TransactionType.TransferAlgo,
+        sign: SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        lsig: escrow,
+        payFlags: { closeRemainderTo: creator.address }
+      }
+    ];
+    await runtime.executeTx(txGroup, program, []);
+    // TODO- close account and tranfer funds to closeRemainderTo
+  });
```

Comment:
Re my comment in Discord. Unit tests depend on each other. So, either put them into one big uni-test (those which depend on each other). Or prepare a clear state before running each unit-test (this is recommended).

Reply:
understood